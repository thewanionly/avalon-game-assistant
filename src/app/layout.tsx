import type { Metadata } from 'next';
import { Geist, Geist_Mono } from 'next/font/google';
import './globals.css';
import { ThemeProvider } from '@/components/theme-provider';
import { Header } from '@/components/Header';

const geistSans = Geist({
  variable: '--font-geist-sans',
  subsets: ['latin'],
});

const geistMono = Geist_Mono({
  variable: '--font-geist-mono',
  subsets: ['latin'],
});

export const metadata: Metadata = {
  title: 'Avalon Game Assistant',
  description:
    'This application provides various tools to assist in playing the Avalon social deduction game.',
};

export default function RootLayout({
  children,
}: Readonly<{
  children: React.ReactNode;
}>) {
  return (
    <html lang="en">
      <body className={`${geistSans.variable} ${geistMono.variable} dark antialiased`}>
        <ThemeProvider
          attribute="class"
          defaultTheme="system"
          enableSystem
          disableTransitionOnChange
        >
          <Header />
          {children}
        </ThemeProvider>
      </body>
    </html>
  );
}
