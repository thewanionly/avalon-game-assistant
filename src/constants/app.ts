export const APP_TITLE = 'Avalon Game Assistant';
