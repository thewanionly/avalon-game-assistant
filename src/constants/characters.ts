import idify from '@/utils/idify';

export enum AvalonCharacterName {
  Merlin = 'Merlin',
  Percival = 'Percival',
  LoyalServantOfArthur = 'Loyal Servant of Arthur',
  Mordred = 'Mordred',
  Morgana = 'Morgana',
  Oberon = 'Oberon',
  Assassin = 'Assassin',
  MinionOfMordred = 'Minion of Mordred',
}

export enum AvalonCharacterLoyalty {
  Good = 'good',
  Evil = 'evil',
}

interface AvalonCharacter {
  id: string;
  name: string;
  uniqueLabel: string;
  loyalty: AvalonCharacterLoyalty;
  isAdditional?: boolean;
  isRequired?: boolean;
  isDefaultChecked?: boolean;
}

const AVALON_CHARACTERS: AvalonCharacter[] = [
  {
    id: idify(AvalonCharacterName.Merlin),
    name: AvalonCharacterName.Merlin,
    uniqueLabel: AvalonCharacterName.Merlin,
    loyalty: AvalonCharacterLoyalty.Good,
    isRequired: true,
    isDefaultChecked: true,
  },
  {
    id: idify(AvalonCharacterName.Percival),
    name: AvalonCharacterName.Percival,
    uniqueLabel: AvalonCharacterName.Percival,
    loyalty: AvalonCharacterLoyalty.Good,
    isAdditional: true,
  },
  {
    id: idify(`${AvalonCharacterName.LoyalServantOfArthur} 1`),
    name: AvalonCharacterName.LoyalServantOfArthur,
    uniqueLabel: `1st ${AvalonCharacterName.LoyalServantOfArthur}`,
    loyalty: AvalonCharacterLoyalty.Good,
    isDefaultChecked: true,
  },
  {
    id: idify(`${AvalonCharacterName.LoyalServantOfArthur} 2`),
    name: AvalonCharacterName.LoyalServantOfArthur,
    uniqueLabel: `2nd ${AvalonCharacterName.LoyalServantOfArthur}`,
    loyalty: AvalonCharacterLoyalty.Good,
    isDefaultChecked: true,
  },
  {
    id: idify(`${AvalonCharacterName.LoyalServantOfArthur} 3`),
    name: AvalonCharacterName.LoyalServantOfArthur,
    uniqueLabel: `3rd ${AvalonCharacterName.LoyalServantOfArthur}`,
    loyalty: AvalonCharacterLoyalty.Good,
  },
  {
    id: idify(`${AvalonCharacterName.LoyalServantOfArthur} 4`),
    name: AvalonCharacterName.LoyalServantOfArthur,
    uniqueLabel: `4th ${AvalonCharacterName.LoyalServantOfArthur}`,
    loyalty: AvalonCharacterLoyalty.Good,
  },
  {
    id: idify(`${AvalonCharacterName.LoyalServantOfArthur} 5`),
    name: AvalonCharacterName.LoyalServantOfArthur,
    uniqueLabel: `5th ${AvalonCharacterName.LoyalServantOfArthur}`,
    loyalty: AvalonCharacterLoyalty.Good,
  },
  {
    id: idify(AvalonCharacterName.Mordred),
    name: AvalonCharacterName.Mordred,
    uniqueLabel: AvalonCharacterName.Mordred,
    loyalty: AvalonCharacterLoyalty.Evil,
    isAdditional: true,
  },
  {
    id: idify(AvalonCharacterName.Morgana),
    name: AvalonCharacterName.Morgana,
    uniqueLabel: AvalonCharacterName.Morgana,
    loyalty: AvalonCharacterLoyalty.Evil,
    isAdditional: true,
  },
  {
    id: idify(AvalonCharacterName.Oberon),
    name: AvalonCharacterName.Oberon,
    uniqueLabel: AvalonCharacterName.Oberon,
    loyalty: AvalonCharacterLoyalty.Evil,
    isAdditional: true,
  },
  {
    id: idify(AvalonCharacterName.Assassin),
    name: AvalonCharacterName.Assassin,
    uniqueLabel: AvalonCharacterName.Assassin,
    loyalty: AvalonCharacterLoyalty.Evil,
    isRequired: true,
    isDefaultChecked: true,
  },
  {
    id: idify(`${AvalonCharacterName.MinionOfMordred} 1`),
    name: AvalonCharacterName.MinionOfMordred,
    uniqueLabel: `1st ${AvalonCharacterName.MinionOfMordred}`,
    loyalty: AvalonCharacterLoyalty.Evil,
    isDefaultChecked: true,
  },
  {
    id: idify(`${AvalonCharacterName.MinionOfMordred} 2`),
    name: AvalonCharacterName.MinionOfMordred,
    uniqueLabel: `2nd ${AvalonCharacterName.MinionOfMordred}`,
    loyalty: AvalonCharacterLoyalty.Evil,
  },
  {
    id: idify(`${AvalonCharacterName.MinionOfMordred} 3`),
    name: AvalonCharacterName.MinionOfMordred,
    uniqueLabel: `3rd ${AvalonCharacterName.MinionOfMordred}`,
    loyalty: AvalonCharacterLoyalty.Evil,
  },
];

export const GOOD_AVALON_CHARACTERS = AVALON_CHARACTERS.filter(
  (character) => character.loyalty === AvalonCharacterLoyalty.Good
);
export const EVIL_AVALON_CHARACTERS = AVALON_CHARACTERS.filter(
  (character) => character.loyalty === AvalonCharacterLoyalty.Evil
);

export const GOOD_REQUIRED_CHARACTERS = GOOD_AVALON_CHARACTERS.filter(
  ({ isRequired }) => isRequired
);
export const EVIL_REQUIRED_CHARACTERS = EVIL_AVALON_CHARACTERS.filter(
  ({ isRequired }) => isRequired
);

export const DEFAULT_GOOD_CHARACTERS = GOOD_AVALON_CHARACTERS.filter(
  ({ isDefaultChecked }) => isDefaultChecked
);
export const DEFAULT_EVIL_CHARACTERS = EVIL_AVALON_CHARACTERS.filter(
  ({ isDefaultChecked }) => isDefaultChecked
);

export const TEAM_DISTRIBUTION = {
  5: {
    [AvalonCharacterLoyalty.Good]: 3,
    [AvalonCharacterLoyalty.Evil]: 2,
  },
  6: {
    [AvalonCharacterLoyalty.Good]: 4,
    [AvalonCharacterLoyalty.Evil]: 2,
  },
  7: {
    [AvalonCharacterLoyalty.Good]: 4,
    [AvalonCharacterLoyalty.Evil]: 3,
  },
  8: {
    [AvalonCharacterLoyalty.Good]: 5,
    [AvalonCharacterLoyalty.Evil]: 3,
  },
  9: {
    [AvalonCharacterLoyalty.Good]: 6,
    [AvalonCharacterLoyalty.Evil]: 3,
  },
  10: {
    [AvalonCharacterLoyalty.Good]: 6,
    [AvalonCharacterLoyalty.Evil]: 4,
  },
};
