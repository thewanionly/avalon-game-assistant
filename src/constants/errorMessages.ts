export const ERROR_REQUIRED_CHARACTERS =
  'You must include the following required characters: {{requiredCharacters}}';
export const ERROR_MIN_PLAYERS =
  'The minimum number of players is {{minPlayers}}. Please add more players.';
export const ERROR_MAX_PLAYERS =
  'The maximum number of players is {{maxPlayers}}. Please reduce the number of players.';
export const ERROR_CHARACTER_DISTRIBUTION = `The character distribution isn't quite right. In a {{numberOfPlayers}}-player game, you need exactly {{goodPlayersCount}} good characters and {{evilPlayersCount}} evil characters to proceed. Please update your selection accordingly.`;
