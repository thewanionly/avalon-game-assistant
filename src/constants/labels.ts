export const GOOD_CHARACTERS_LABEL = 'Good characters ({{count}})';
export const GOOD_CHARACTERS_NO_SELECTED_LABEL = 'Good characters';
export const EVIL_CHARACTERS_LABEL = 'Evil characters ({{count}})';
export const EVIL_CHARACTERS_NO_SELECTED_LABEL = 'Evil characters';
export const PLAY_BUTTON_LABEL = 'Play {{count}}';
export const PLAY_BUTTON_NO_SELECTED_LABEL = 'Play';
