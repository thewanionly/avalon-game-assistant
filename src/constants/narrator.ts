export enum NarratorStatus {
  IDLE = 'idle',
  PLAYING = 'playing',
  PAUSED = 'paused',
  END = 'end',
}
