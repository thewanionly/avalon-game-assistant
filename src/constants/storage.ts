export const SELECTED_CHARACTERS_COOKIE = 'selected-characters';
export const VOICE_RATE = 'voice-rate';
