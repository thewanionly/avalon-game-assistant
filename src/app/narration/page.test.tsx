import { render, screen } from '@testing-library/react';

import { NARRATION } from '@/constants/app';
import Page from '@/app/narration/page';

describe('Narration Page', () => {
  it('renders narration title heading', () => {
    render(<Page />);

    const narrationTitle = screen.getByRole('heading', { name: NARRATION.pageTitle, level: 1 });

    expect(narrationTitle).toBeInTheDocument();
  });

  it('renders narration page unchanged', () => {
    const { container } = render(<Page />);
    expect(container).toMatchSnapshot();
  });
});
