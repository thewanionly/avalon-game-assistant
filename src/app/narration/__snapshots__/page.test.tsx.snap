// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Narration Page renders narration page unchanged 1`] = `
<div>
  <div
    class="mx-auto max-w-3xl px-4 py-8 font-[family-name:var(--font-geist-sans)] sm:p-20"
  >
    <h1
      class="mb-12 text-2xl"
    >
      Narration
    </h1>
    <form
      class="flex flex-col justify-between gap-6 *:"
    >
      <div
        class="space-y-3"
      >
        <label
          class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
          for=":rg:-form-item"
        >
          Good characters (3)
        </label>
        <div
          class="grid grid-cols-[repeat(auto-fit,_minmax(80px,_1fr))] gap-1"
        >
          <div
            class="flex items-center gap-2 space-y-0"
          >
            <label
              class="w-full"
            >
              <input
                aria-checked="true"
                aria-describedby=":rh:-form-item-description"
                aria-invalid="false"
                aria-label="Merlin"
                checked=""
                class="peer absolute m-0 appearance-none"
                id=":rh:-form-item"
                required=""
                type="checkbox"
              />
              <div
                class="flex aspect-[3/4] items-center justify-center rounded-md p-4 text-center text-sm bg-blue-500 text-neutral-900 w-full peer-focus:outline peer-focus:outline-2 peer-focus:outline-gray-500 peer-focus:ring-2 peer-focus:ring-gray-300 after:ml-0.5 after:content-['*']"
              >
                Merlin
              </div>
              <span
                class="sr-only text-sm font-normal after:ml-0.5 after:text-red-500 after:content-['*']"
              >
                Merlin
              </span>
            </label>
          </div>
          <div
            class="flex items-center gap-2 space-y-0"
          >
            <label
              class="w-full"
            >
              <input
                aria-checked="false"
                aria-describedby=":ri:-form-item-description"
                aria-invalid="false"
                aria-label="Percival"
                class="peer absolute m-0 appearance-none"
                id=":ri:-form-item"
                type="checkbox"
              />
              <div
                class="flex aspect-[3/4] items-center justify-center rounded-md p-4 text-center text-sm bg-blue-500/50 text-neutral-900/50 w-full peer-focus:outline peer-focus:outline-2 peer-focus:outline-gray-500 peer-focus:ring-2 peer-focus:ring-gray-300"
              >
                Percival
              </div>
              <span
                class="sr-only text-sm font-normal"
              >
                Percival
              </span>
            </label>
          </div>
          <div
            class="flex items-center gap-2 space-y-0"
          >
            <label
              class="w-full"
            >
              <input
                aria-checked="true"
                aria-describedby=":rj:-form-item-description"
                aria-invalid="false"
                aria-label="1st Loyal Servant of Arthur"
                checked=""
                class="peer absolute m-0 appearance-none"
                id=":rj:-form-item"
                type="checkbox"
              />
              <div
                class="flex aspect-[3/4] items-center justify-center rounded-md p-4 text-center text-sm bg-blue-500 text-neutral-900 w-full peer-focus:outline peer-focus:outline-2 peer-focus:outline-gray-500 peer-focus:ring-2 peer-focus:ring-gray-300"
              >
                Loyal Servant of Arthur
              </div>
              <span
                class="sr-only text-sm font-normal"
              >
                Loyal Servant of Arthur
              </span>
            </label>
          </div>
          <div
            class="flex items-center gap-2 space-y-0"
          >
            <label
              class="w-full"
            >
              <input
                aria-checked="true"
                aria-describedby=":rk:-form-item-description"
                aria-invalid="false"
                aria-label="2nd Loyal Servant of Arthur"
                checked=""
                class="peer absolute m-0 appearance-none"
                id=":rk:-form-item"
                type="checkbox"
              />
              <div
                class="flex aspect-[3/4] items-center justify-center rounded-md p-4 text-center text-sm bg-blue-500 text-neutral-900 w-full peer-focus:outline peer-focus:outline-2 peer-focus:outline-gray-500 peer-focus:ring-2 peer-focus:ring-gray-300"
              >
                Loyal Servant of Arthur
              </div>
              <span
                class="sr-only text-sm font-normal"
              >
                Loyal Servant of Arthur
              </span>
            </label>
          </div>
          <div
            class="flex items-center gap-2 space-y-0"
          >
            <label
              class="w-full"
            >
              <input
                aria-checked="false"
                aria-describedby=":rl:-form-item-description"
                aria-invalid="false"
                aria-label="3rd Loyal Servant of Arthur"
                class="peer absolute m-0 appearance-none"
                id=":rl:-form-item"
                type="checkbox"
              />
              <div
                class="flex aspect-[3/4] items-center justify-center rounded-md p-4 text-center text-sm bg-blue-500/50 text-neutral-900/50 w-full peer-focus:outline peer-focus:outline-2 peer-focus:outline-gray-500 peer-focus:ring-2 peer-focus:ring-gray-300"
              >
                Loyal Servant of Arthur
              </div>
              <span
                class="sr-only text-sm font-normal"
              >
                Loyal Servant of Arthur
              </span>
            </label>
          </div>
          <div
            class="flex items-center gap-2 space-y-0"
          >
            <label
              class="w-full"
            >
              <input
                aria-checked="false"
                aria-describedby=":rm:-form-item-description"
                aria-invalid="false"
                aria-label="4th Loyal Servant of Arthur"
                class="peer absolute m-0 appearance-none"
                id=":rm:-form-item"
                type="checkbox"
              />
              <div
                class="flex aspect-[3/4] items-center justify-center rounded-md p-4 text-center text-sm bg-blue-500/50 text-neutral-900/50 w-full peer-focus:outline peer-focus:outline-2 peer-focus:outline-gray-500 peer-focus:ring-2 peer-focus:ring-gray-300"
              >
                Loyal Servant of Arthur
              </div>
              <span
                class="sr-only text-sm font-normal"
              >
                Loyal Servant of Arthur
              </span>
            </label>
          </div>
          <div
            class="flex items-center gap-2 space-y-0"
          >
            <label
              class="w-full"
            >
              <input
                aria-checked="false"
                aria-describedby=":rn:-form-item-description"
                aria-invalid="false"
                aria-label="5th Loyal Servant of Arthur"
                class="peer absolute m-0 appearance-none"
                id=":rn:-form-item"
                type="checkbox"
              />
              <div
                class="flex aspect-[3/4] items-center justify-center rounded-md p-4 text-center text-sm bg-blue-500/50 text-neutral-900/50 w-full peer-focus:outline peer-focus:outline-2 peer-focus:outline-gray-500 peer-focus:ring-2 peer-focus:ring-gray-300"
              >
                Loyal Servant of Arthur
              </div>
              <span
                class="sr-only text-sm font-normal"
              >
                Loyal Servant of Arthur
              </span>
            </label>
          </div>
        </div>
      </div>
      <div
        class="space-y-3"
      >
        <label
          class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
          for=":ro:-form-item"
        >
          Evil characters (2)
        </label>
        <div
          class="grid grid-cols-[repeat(auto-fit,_minmax(80px,_1fr))] gap-1"
        >
          <div
            class="flex items-center gap-2 space-y-0"
          >
            <label
              class="w-full"
            >
              <input
                aria-checked="false"
                aria-describedby=":rp:-form-item-description"
                aria-invalid="false"
                aria-label="Mordred"
                class="peer absolute m-0 appearance-none"
                id=":rp:-form-item"
                type="checkbox"
              />
              <div
                class="flex aspect-[3/4] items-center justify-center rounded-md p-4 text-center text-sm bg-red-500/50 text-slate-50/50 w-full peer-focus:outline peer-focus:outline-2 peer-focus:outline-gray-500 peer-focus:ring-2 peer-focus:ring-gray-300"
              >
                Mordred
              </div>
              <span
                class="sr-only text-sm font-normal"
              >
                Mordred
              </span>
            </label>
          </div>
          <div
            class="flex items-center gap-2 space-y-0"
          >
            <label
              class="w-full"
            >
              <input
                aria-checked="false"
                aria-describedby=":rq:-form-item-description"
                aria-invalid="false"
                aria-label="Morgana"
                class="peer absolute m-0 appearance-none"
                id=":rq:-form-item"
                type="checkbox"
              />
              <div
                class="flex aspect-[3/4] items-center justify-center rounded-md p-4 text-center text-sm bg-red-500/50 text-slate-50/50 w-full peer-focus:outline peer-focus:outline-2 peer-focus:outline-gray-500 peer-focus:ring-2 peer-focus:ring-gray-300"
              >
                Morgana
              </div>
              <span
                class="sr-only text-sm font-normal"
              >
                Morgana
              </span>
            </label>
          </div>
          <div
            class="flex items-center gap-2 space-y-0"
          >
            <label
              class="w-full"
            >
              <input
                aria-checked="false"
                aria-describedby=":rr:-form-item-description"
                aria-invalid="false"
                aria-label="Oberon"
                class="peer absolute m-0 appearance-none"
                id=":rr:-form-item"
                type="checkbox"
              />
              <div
                class="flex aspect-[3/4] items-center justify-center rounded-md p-4 text-center text-sm bg-red-500/50 text-slate-50/50 w-full peer-focus:outline peer-focus:outline-2 peer-focus:outline-gray-500 peer-focus:ring-2 peer-focus:ring-gray-300"
              >
                Oberon
              </div>
              <span
                class="sr-only text-sm font-normal"
              >
                Oberon
              </span>
            </label>
          </div>
          <div
            class="flex items-center gap-2 space-y-0"
          >
            <label
              class="w-full"
            >
              <input
                aria-checked="true"
                aria-describedby=":rs:-form-item-description"
                aria-invalid="false"
                aria-label="Assassin"
                checked=""
                class="peer absolute m-0 appearance-none"
                id=":rs:-form-item"
                required=""
                type="checkbox"
              />
              <div
                class="flex aspect-[3/4] items-center justify-center rounded-md p-4 text-center text-sm bg-red-500 text-slate-50 w-full peer-focus:outline peer-focus:outline-2 peer-focus:outline-gray-500 peer-focus:ring-2 peer-focus:ring-gray-300 after:ml-0.5 after:content-['*']"
              >
                Assassin
              </div>
              <span
                class="sr-only text-sm font-normal after:ml-0.5 after:text-red-500 after:content-['*']"
              >
                Assassin
              </span>
            </label>
          </div>
          <div
            class="flex items-center gap-2 space-y-0"
          >
            <label
              class="w-full"
            >
              <input
                aria-checked="true"
                aria-describedby=":rt:-form-item-description"
                aria-invalid="false"
                aria-label="1st Minion of Mordred"
                checked=""
                class="peer absolute m-0 appearance-none"
                id=":rt:-form-item"
                type="checkbox"
              />
              <div
                class="flex aspect-[3/4] items-center justify-center rounded-md p-4 text-center text-sm bg-red-500 text-slate-50 w-full peer-focus:outline peer-focus:outline-2 peer-focus:outline-gray-500 peer-focus:ring-2 peer-focus:ring-gray-300"
              >
                Minion of Mordred
              </div>
              <span
                class="sr-only text-sm font-normal"
              >
                Minion of Mordred
              </span>
            </label>
          </div>
          <div
            class="flex items-center gap-2 space-y-0"
          >
            <label
              class="w-full"
            >
              <input
                aria-checked="false"
                aria-describedby=":ru:-form-item-description"
                aria-invalid="false"
                aria-label="2nd Minion of Mordred"
                class="peer absolute m-0 appearance-none"
                id=":ru:-form-item"
                type="checkbox"
              />
              <div
                class="flex aspect-[3/4] items-center justify-center rounded-md p-4 text-center text-sm bg-red-500/50 text-slate-50/50 w-full peer-focus:outline peer-focus:outline-2 peer-focus:outline-gray-500 peer-focus:ring-2 peer-focus:ring-gray-300"
              >
                Minion of Mordred
              </div>
              <span
                class="sr-only text-sm font-normal"
              >
                Minion of Mordred
              </span>
            </label>
          </div>
          <div
            class="flex items-center gap-2 space-y-0"
          >
            <label
              class="w-full"
            >
              <input
                aria-checked="false"
                aria-describedby=":rv:-form-item-description"
                aria-invalid="false"
                aria-label="3rd Minion of Mordred"
                class="peer absolute m-0 appearance-none"
                id=":rv:-form-item"
                type="checkbox"
              />
              <div
                class="flex aspect-[3/4] items-center justify-center rounded-md p-4 text-center text-sm bg-red-500/50 text-slate-50/50 w-full peer-focus:outline peer-focus:outline-2 peer-focus:outline-gray-500 peer-focus:ring-2 peer-focus:ring-gray-300"
              >
                Minion of Mordred
              </div>
              <span
                class="sr-only text-sm font-normal"
              >
                Minion of Mordred
              </span>
            </label>
          </div>
        </div>
      </div>
      <div
        class="sticky bottom-0 bg-white pb-5 pt-7"
      >
        <button
          class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-primary text-primary-foreground shadow hover:bg-primary/90 h-9 px-4 py-2 w-full"
          type="submit"
        >
          Play 5
        </button>
      </div>
    </form>
  </div>
</div>
`;
